src/test/java/step3/application/GameDirectorTest.java

```diff
@@ -0,0 +1,42 @@
+package step3.application;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import step3.domain.board.GameBoard;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class GameDirectorTest {
+
+    private static final int FIRST_ROW_INDEX = 0;
+
+    @DisplayName("n대의 자동차가 각각 주어진 횟수 만큼 정지 or 전진 하는지 테스트")
+    @Test
+    void playGameTest() {
+        //given
+        GameBoard gameBoard = new GameBoard();
+        GameDirector gameDirector = new GameDirector(gameBoard);
+
+        int carCount = 5;
+        int roundSize = 10;
+
+        //when
+        gameDirector.playGame(carCount, roundSize);
```

Comment:
```suggestion
        int round = 0;

        //when
        while(gameDirector.hasNextRound()) {
            round++;
            gameDirector.racing(carCount, roundSize);
        }
```
예를 들면 이런식으로 경주가 진행되는 횟수를 확인해보면 어떨까요?

지금은 경주가 진행되었다는 것을 확인하기 위해 GameBoard의 readAllRecords()를 사용해야되는데
제시한 코드처럼 변경이 된다면 GameBoard를 사용하지 않은 형태로 설계 내용을 바꾸어 볼 수 도 있지 않을까요. 🤔

Reply:
이 부분은 좀 더 고민해볼게요 🙇

Reply:
```
while(gameDirector.hasNextRound()) {
            round++;
            gameDirector.racing(carCount, roundSize);
}
```
처럼 진행할 경우 racing() 전에 hasNextRound() 가 실행되야하는 일종의 규칙같은게 생길 것 같은데요.

두 메서드가 public 으로 되어있을거고 이 때,  
hasNextRound()로 다음 라운드가 있는지 확인하지 않고 racing()을 호출 했을 때
더이상 남은 라운드가 없음에도 불구하고 라운드가 진행되는 비정상적인 상황이 생길 것 같아요.

그렇다고 racing()메소드에 다음 라운드가 있는지 검증하는 부분을 추가한다면
hasNextRound() 이후에 racing()를 호출하는 경우 `다음라운드가 있는지` 검증이 중복해서 발생할 것 같아요.

꼭 지금미션이 아니더라도 이부분은 제가 항상 하는 고민인데요.
저는 위에서 제가 말씀드린 이유로 제안해주신 `객체에게 메소드를 수행할 수 있는지 묻고 수행시키기` 로직을 좀 꺼려하는 편인데
용대님은 어떻게 생각하시나요? 
항상 고민하던 부분이라 마침 질문할 기회가 생겨서 다행이네요 👍 

> GameBoard를 사용하지 않은 형태로 설계 내용을 바꾸어 볼 수 도 있지 않을까요.

GameBoard를 사용하지 않는 형태는 고민해봤는데 쉽지 않네요..
좀 더 고민해볼게요!

Reply:
`객체에게 메소드를 수행할 수 있는지 묻고 수행시키기` 방법이 좋지 않은 것은 맞는 이야기에요.
제시한 코드 내용도 최대한 지금 설계 내용에 맞추어서 드린 부분이다보니 디미터의 법칙을 적용할수 없었습니다. 😢

다만 `주어진 횟수 동안`에 대한 테스트를 위해 두개의 GameBoard, GameDirector 객체를 이용하는 것보다
하나의 객체를 이용해서 할 수 있는 방법에 대한 고민을 위해 드린 의견이었어요.

        GameDirector gameDirector = new GameDirector(1, 1);
        gameDirector.race();
        assertThat(gameDirector.isFinish()).isTrue();

형태는 이런 형태가 되어도 상관이 없어요. 앞서 이야기드린 것과 같이 `주어진 횟수 동안`이라는 테스트를
하나의 객체로 할 수 있게 변경해보면 좋겠습니다. :)

Reply:
`객체에게 메소드를 수행할 수 있는지 묻고 수행시키기` 방법이 좋지 않은 것은 맞는 이야기에요.
제시한 코드 내용도 최대한 지금 설계 내용에 맞추어서 드린 부분이다보니 디미터의 법칙을 적용할수 없었습니다. 😢

다만 `주어진 횟수 동안`에 대한 테스트를 위해 두개의 GameBoard, GameDirector 객체를 이용하는 것보다
하나의 객체를 이용해서 할 수 있는 방법에 대한 고민을 위해 드린 의견이었어요.

        GameDirector gameDirector = new GameDirector(1, 1);
        gameDirector.race();
        assertThat(gameDirector.isFinish()).isTrue();

형태는 이런 형태가 되어도 상관이 없어요. 앞서 이야기드린 것과 같이 `주어진 횟수 동안`이라는 테스트를
하나의 객체로 할 수 있게 변경해보면 좋겠습니다. :)